### PR TITLE
Create Dev-Process.md

### PR DESCRIPTION
**Pull request by cjm721**
Added some basic development process requirements. This will be expanded as needed.

### CODE REVIEW

File: Dev-Process.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+Master needs to always be in a releasable state. To ensure this each feature will be in 
+its own branch and each developer will commit to their own branch that is based on the 
+feature branch they are working on.
+
+### Naming Example
+* master
+  * f_[Feature_Name]
+    * f_[Feature_name]_[user]
+
```
Comment (shellajt) [inline Dev-Process.md]: comment

File: Dev-Process.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+All methods and structures should be named in a manner that gives simple info about what that element is for.
+
+### Testing
+
+**Code must be written so that it is testable.** All non-private methods should be have some form of unit tests (pure UI methods are the exception).
+There should be Integration and System tests but those will be decided on a feature basis.
+
+### WhiteSpace
+
+There should be no tab character in any of the source code. It should be either 2 or 4 spaces.
```
Comment (shellajt) [inline Dev-Process.md]: Do we want to limit ourselves to either 2 or 4? Consistency within our codebase would be nice

Comment (xxshadowfang) [reply Dev-Process.md]: I think we should specify one standard as well. I like 4 spaces

**Comment by shellajt** [comment]
We need to find a way to lock down master so we cant push to it or merge PRs that havn't been approved

File: Dev-Process.md (hunk 2 above)
Comment (taylerhow) [reply Dev-Process.md]: I agree. I prefer 4 spaces.

**Comment by cjm721** [comment]
Master is already protected so a force push cannot be applied but there are ways to lock it even better.

See [status](https://developer.github.com/v3/repos/statuses/)
And [Integrations](https://github.com/integrations)

**Comment by shellajt** [comment]
Lets squash commits before we merge. Might be worth putting into the doc as well

**Comment by cjm721** [comment]
If we squash commits though we lost direct access to the history. So makes looking up specific times very hard. 

If you just don't like the merge commit (which I am not much of a fan of either for small commit groups)  a Fast Forward Merge takes care of that.

I can edit your comment? - Jesse

**Comment by shellajt** [comment]
In general, sure. But for something like this were you made an edit to take out two words...

We can take out the tiny commits that arn't important. Small fixes, formatting, etc

**Comment by taylerhow** [comment]
Looks good!

A few technicalities:
- Friday stand-ups are at 3:30, I believe.
- Are we having Wednesday stand-ups (in addition to meetings w/ Sriram, Sean, retros, and grooming)?

**Comment by richarel** [comment]
Looks good to me too. I'll second everything Tayler said regarding the meeting times ^.

**Review by shellajt — APPROVED** [review]

**Pull request merged**